### PR TITLE
define `Base.isstored` for diagonal

### PR DESCRIPTION
**Pull request by longemen3000**
required to fix https://github.com/JuliaLang/julia/issues/50377

### CODE REVIEW

**Comment by longemen3000** [comment]
the tests are really broken?

**Comment by dkarrasch** [comment]
You need to make the PR in Julia Base.

File: src/LinearAlgebra.jl [hunk 1]
```diff
@@ -11,7 +11,7 @@ import Base: \, /, *, ^, +, -, ==
 import Base: USE_BLAS64, abs, acos, acosh, acot, acoth, acsc, acsch, adjoint, asec, asech,
     asin, asinh, atan, atanh, axes, big, broadcast, ceil, cis, conj, convert, copy, copyto!,
     copymutable, cos, cosh, cot, coth, csc, csch, eltype, exp, fill!, floor, getindex, hcat,
-    getproperty, imag, inv, isapprox, isequal, isone, iszero, IndexStyle, kron, kron!,
+    getproperty, imag, inv, isapprox, isequal, isone, isstored, iszero, IndexStyle, kron, kron!,
```
Comment (stevengj) [inline src/LinearAlgebra.jl:14]: I think it's clearer just to define `Base.isstored` I think.

**Comment by longemen3000** [comment]
closed in favor of https://github.com/JuliaLang/julia/pull/50391

**Pull request closed without merging**